pkg/clusterapi/apibuilder.go

```diff
@@ -133,7 +133,7 @@ func KubeadmControlPlane(clusterSpec *cluster.Spec, infrastructureObject APIObje
 			Kind:       kubeadmControlPlaneKind,
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      KubeadmControlPlaneName(clusterSpec),
+			Name:      KubeadmControlPlaneName(clusterSpec.Cluster),
```

Comment:
Why do we need changes under clusterapi?

Reply:
This was so I can use the `KubeadmControlPlaneName()` function in `GetKubeadmControlPlane` with the cluster instead of requiring a clusterSpec: https://github.com/aws/eks-anywhere/pull/4994/files#diff-11f996f58e1db6221bc9e3ea4a62440e4071227bf110ecf71efb26024c34218cR49

Just like `GetCAPICluster()`